spec/services/checksum_validator_spec.rb

```diff
@@ -203,24 +203,60 @@
         expect(results).to receive(:add_result).with(
           AuditResults::INVALID_MANIFEST, manifest_file_path: 'spec/fixtures/checksum_root01/moab_storage_trunk/xx/444/xx/4444/xx444xx4444/v0001/manifests/signatureCatalog.xml'
         )
-        cv.validate_signature_catalog
+        cv.send(:validate_signature_catalog_listing)
       end
     end
   end
 
-  context '#validate_checksum' do
+  context '#flag_unexpected_data_content_files' do
     let(:druid) { 'bj102hs9687' }
-    let(:cv) { described_class.new(druid, 'fixture_sr1') }
+    let(:cv) { described_class.new(druid, endpoint_name) }
 
-    it 'returns a positive result for a druid that passes validation' do
-      cv.validate_checksum
-      expect(cv.checksum_results.result_array.first).to have_key(:moab_checksum_valid)
+    it 'calls validate_against_signature_catalog on each of the data_content_files' do
+      files = ['spec/fixtures/storage_root01/moab_storage_trunk/bj/102/hs/9687/bj102hs9687/v0001/data/content/eric-smith-dissertation-augmented.pdf',
+               'spec/fixtures/storage_root01/moab_storage_trunk/bj/102/hs/9687/bj102hs9687/v0001/data/content/eric-smith-dissertation.pdf']
+      expect(cv).to receive(:data_content_files).and_return(files)
+      files.each do |file|
+        expect(cv).to receive(:validate_against_signature_catalog).with(file)
+      end
+      cv.send(:flag_unexpected_data_content_files)
     end
 
-    it 'returns error codes for a druid that fails validation' do
-      cv = described_class.new(druid, endpoint_name)
-      cv.validate_checksum
-      expect(cv.checksum_results.result_array.first).to have_key(:file_not_in_manifest)
+    it 'calls AuditResults.report_results' do
+      results = instance_double(AuditResults, add_result: nil, :actual_version= => nil, :check_name= => nil)
+      allow(AuditResults).to receive(:new).and_return(results)
+      expect(results).to receive(:report_results)
+      cv.send(:flag_unexpected_data_content_files)
+    end
+
+    context 'file is on disk, but not present in signatureCatalog.xml' do
+      it 'adds a FILE_NOT_IN_MANIFEST error' do
+        druid = 'zz555zz5555'
+        file_path = 'spec/fixtures/checksum_root01/moab_storage_trunk/zz/555/zz/5555/zz555zz5555/v0001/data/content/not_in_sigcat.txt'
+        manifest_file_path = 'spec/fixtures/checksum_root01/moab_storage_trunk/zz/555/zz/5555/zz555zz5555/v0002/manifests/signatureCatalog.xml'
+        results = instance_double(AuditResults, report_results: nil, check_name: nil)
+        allow(AuditResults).to receive(:new).and_return(results)
+        cv = described_class.new(druid, endpoint_name)
+        expect(results).to receive(:add_result).with(
+          AuditResults::FILE_NOT_IN_MANIFEST, file_path: file_path, manifest_file_path: manifest_file_path
+        )
+        cv.send(:flag_unexpected_data_content_files)
+      end
```

Comment:
I would like to see more tests here:
- for a metadata file
- for a file in a content subdir
- ...?